### PR TITLE
Quickfix updates

### PR DESCRIPTION
**Pull request by 333fred**
Simplifies a few places we can replace with `?.` invocation and throw expressions. Also annotates the auto-generated code to suppress the use throw invocation message, as it warns 91 times from that code. Depends on #709.
@dotnet/project-system

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet/project-system this is ready for review, when you get a few minutes.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj [hunk 1]
```diff
@@ -88,6 +88,9 @@
     <Compile Include="ProjectSystem\Rules\AnalyzerReference.xaml.cs">
       <DependentUpon>AnalyzerReference.xaml</DependentUpon>
     </Compile>
+    <Compile Include="ProjectSystem\Rules\ProjectDebugger.xaml.cs">
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj]: Why this change? Was this missed out before?

Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj]: Or is this a file that we used to have but don't need anymore after the move to launch profiles? If so, we should delete it. Either way, can you check it's setupauthoring as well?

Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj]: Yes. Several of these were either not there, or had the wrong name (like in DebuggerGeneral).

Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj]: For the setup authoring, it doesn't look like we deploy the .cs files at all. They're just there for IDE features, like the suppressions and turning off code coverage.
